### PR TITLE
RATIS-1892 Unify the lifetime of the RaftServerProxy thread pool

### PR DESCRIPTION
**Pull request by OneSizeFitsQuorum**
see [jira](https://issues.apache.org/jira/browse/RATIS-1892)

### CODE REVIEW

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java [hunk 1]
```diff
@@ -416,6 +416,12 @@ public void close() {
       LOG.warn(getId() + ": Failed to shutdown implExecutor", ignored);
     }
 
+    try {
+      ConcurrentUtils.shutdownAndWait(executor);
+    } catch (Exception ignored) {
+      LOG.warn(getId() + ": Failed to shutdown executor", ignored);
+    }
+
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java:424]: We should move `implExecutor` inside `checkStateAndClose(..)` instead.  And also initialize `implExecutor` and `executor` in the `startImpl()` method.

**Review by szetszwo — COMMENTED** [review]
@OneSizeFitsQuorum , thanks a lot for working on this!  Please see the comment inlined and also https://issues.apache.org/jira/secure/attachment/13063032/923_review.patch

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java (hunk 1 above)
Comment (OneSizeFitsQuorum) [reply ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java]: Great Idea! Has fixed

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**